### PR TITLE
move transaction logic to service

### PR DESCRIPTION
**Pull request by MariaPadalka**

### CODE REVIEW

File: ASPNET-CORE-MVC/Domain/Services/TransactionService/ITransactionService.cs [hunk 1]
```diff
@@ -11,5 +11,5 @@ public interface ITransactionService
 
     //Task<Transaction> UpdateTransaction(TransactionDto category);
 
-    Task RemoveTransaction(Guid categoryId);
+    Task RemoveTransaction(Guid categoryId, Guid userId);
```
Comment (HoshkoS) [inline ASPNET-CORE-MVC/Domain/Services/TransactionService/ITransactionService.cs]: Shouldn`t it be (Guid transaction_id, ...)?

File: ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs [hunk 2]
```diff
@@ -2,87 +2,101 @@
...
+        this._unitOfWork = unitOfWork;
+        this._logger = logger;
+        this._flashMessage = flashMessage;
     }
 
     public async Task<ICollection<TransactionDto>> GetUserTransactions(Guid UserId)
     {
-        try
+        var transactions = await this._unitOfWork.Transactions.Find(c => c.UserId == UserId);
+        if (transactions == null)
```
Comment (HoshkoS) [inline ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs]: If I am not mistaken, Find always returns Collection and will just return empty collection with count=0, and this block won`t be executed

Comment (MariaPadalka) [reply ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs]: Yes, You are not mistaken. My bad. Fixed. You should consider fixing that mistake in your code too
image

File: ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs [hunk 3]
```diff
@@ -2,87 +2,94 @@
...
-            UserId = transaction.UserId,
-            FromCategory = transaction.FromCategory,
-            ToCategory = transaction.ToCategory,
-            Amount = transaction.Amount,
-            Date = transaction.Date,
-            Details = transaction.Details,
-            Planned = transaction.Planned
-        };
-        await _transactionRepository.Add(newTransaction);
+            var message = "Error occured while deleting transaction";
```
Comment (HoshkoS) [inline ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs]: This message does not match method`s action.

**Review by wehfis — APPROVED** [review]

**Review by HoshkoS — APPROVED** [review]

**Pull request merged**